src/test/codewhisperer/tracker/codewhispererCodeCoverageTracker.test.ts

```diff
@@ -7,28 +7,157 @@ import * as assert from 'assert'
 import * as sinon from 'sinon'
 import * as vscode from 'vscode'
 import { CodeWhispererCodeCoverageTracker } from '../../../codewhisperer/tracker/codewhispererCodeCoverageTracker'
-import { resetCodeWhispererGlobalVariables } from '../testUtil'
+import { createMockDocument, createMockTextEditor, resetCodeWhispererGlobalVariables } from '../testUtil'
 import globals from '../../../shared/extensionGlobals'
 import { assertTelemetryCurried } from '../../testUtil'
+import { vsCodeState } from '../../../codewhisperer/models/model'
 
 describe('codewhispererCodecoverageTracker', function () {
-    const language = 'plaintext'
+    const language = 'python'
     const mockGlobalStorage: vscode.Memento = {
         update: sinon.spy(),
         get: sinon.stub().returns(true),
     }
 
-    describe('setAcceptedTokens', function () {
+    describe('updateAcceptedTokensCount', function () {
         beforeEach(function () {
             resetCodeWhispererGlobalVariables()
             CodeWhispererCodeCoverageTracker.instances.delete(language)
         })
 
-        it('Should set Accepted Tokens when CodeWhisperer terms accepted', function () {
-            const suggestion = "print('Hello World!')"
+        it('Should compute edit distance to update the accepted tokens', function () {
+            const editor = createMockTextEditor('import math\ndef addTwoNumbers(a, b):\n')
             const tracker = CodeWhispererCodeCoverageTracker.getTracker(language, mockGlobalStorage)
-            tracker.setAcceptedTokens(suggestion)
-            assert.strictEqual(tracker.AcceptedTokensLength, 21)
+            tracker?.addAcceptedTokens(editor.document.fileName, {
+                range: new vscode.Range(0, 0, 0, 39),
+                text: `import math\ndef two_sum(nums, target):\n`,
+                accepted: 39,
+            })
+            tracker?.addTotalTokens(editor.document.fileName, 100)
+            tracker?.updateAcceptedTokensCount(editor)
+            assert.strictEqual(tracker?.acceptedTokens[editor.document.fileName][0].accepted, 12)
```

Comment:
Maybe I'm misunderstanding things but I'm confused as to why this is 12?
Distance is 17 so shouldn't it be 22?

Reply:
+1

Reply:
since the test case is regarding edit-distance, could you add a code comment on it what's the user modification here

Reply:
The accepted code is `import math\ndef two_sum(nums, target):\n`

User changed it to 'import math\ndef addTwoNumbers(a, b):\n'

In this case, user accepted 'import math\ndef ', so the accepted is 16. @Will-ShaoHua we can go over this

Reply:
+1, some explaining here could help

Reply:
oops sorry didn't see that

Reply:
just a nit, maybe we can add variables like`originalRecommendation` `modifiedRecommendation`

Reply:
as for the number, i obtained 24.. lol

Reply:
so given this formula https://github.com/aws/aws-toolkit-vscode/pull/2845/files#diff-03e0a75d43072c8dd392c13a8ce7549b075fd4caf70c2539f06fcaee2cc7424dR74, editDistance = 17 and maxOf(originalRecomm.length, modifiedRecomm.length) = maxOf(41, 39)
so the delta should be 41 - 17 = 24

Reply:
I would suggest we can use a simpler originalRecomm & modifiedRecomm and add complicate ones to another test case as mentioned in this thread https://github.com/aws/aws-toolkit-vscode/pull/2845/files#r952992751

Reply:
This test case has been updated. Thanks @Will-ShaoHua